ebean-ddl-generator/src/main/java/io/ebeaninternal/dbmigration/ddlgeneration/platform/PlatformDdl.java

```diff
@@ -20,12 +20,17 @@
 
 import java.util.List;
 import java.util.Locale;
+import java.util.regex.Matcher;
+import java.util.regex.Pattern;
 
 /**
  * Controls the DDL generation for a specific database platform.
  */
 public class PlatformDdl {
 
+  // matches on pattern "check ( COLUMNAME ... )". ColumnName is match group 2;
+  private static final Pattern CHECK_PATTERN = Pattern.compile("(.*?\\( *)([^ ]+)(.*)");
```

Comment:
unfortunately, we must add also quotes to the check-patterns, if we want to support `allQuotedIdentifiers`.
So we have to parse `check ( "mycolumn" in ('A','B'))` and replace the quote chars by the platform specific.
Note: the check constraint is in the model.xml and is the same for all platforms